src/analytics/src/main/scala/osmesa/analytics/oneoffs/ChangesetORCCreator.scala

```diff
@@ -0,0 +1,204 @@
+        .validate("orc must be an S3 or file Uri") { uri =>
+          uri.getScheme.startsWith("s3") || uri.getScheme.startsWith("file")
+        }
+        .validate("orc must be an .orc file") { _.getPath.endsWith(".orc") }
+
+      val outputArg = Opts.argument[URI]("destination ORC")
+        .validate("Output URI must have a scheme") { _.getScheme != null }
+        .validate("Output URI must have an S3 or file scheme") { uri =>
+          uri.getScheme.startsWith("s3") || uri.getScheme.startsWith("file")
+        }
+        .validate("orc must be an .orc file") { _.getPath.endsWith(".orc") }
+
+      (changesetSourceOpt,
+       endTimeOpt,
+       orcArg,
+       outputArg).mapN {
+        (changesetSource, endTime, orcUri, outputURI) =>
+        implicit val spark: SparkSession = Analytics.sparkSession("ChangesetORCCreator")
+
+        import spark.implicits._
+
+        val df = spark.read.orc(orcUri.toString)
+        val lastModified = df.select(max(coalesce('closed_at, 'created_at))).first.getAs[Timestamp](0)
+
+        val startSequence = findSequenceFor(lastModified.toInstant, changesetSource)
+        val endSequence = endTime.map(findSequenceFor(_, changesetSource)).getOrElse(getCurrentSequence(changesetSource).sequence)
+
+        val options = Map(
+          Source.BaseURI -> changesetSource.toString,
+          Source.StartSequence -> startSequence.toString,
+          Source.EndSequence -> (endSequence + 1).toString // sequence range is (]; end sequence is exclusive
+        )
+
+        val changesets = spark.read.format(Source.Changesets).options(options).load
+        changesets
+          .drop("comments", "sequence")
+          .union(df.select(
+            'id,
+            'tags,
+            'created_at as 'createdAt,
+            'open,
+            'closed_at as 'closedAt,
+            'comments_count as 'commentsCount,
+            'min_lat as 'minLat,
+            'max_lat as 'maxLat,
+            'min_lon as 'minLon,
+            'max_lon as 'maxLon,
+            'num_changes as 'numChanges,
+            'uid,
+            'user)
+          )
+          .repartition(1)
+          .write
+          .orc(outputURI.toString)
+
+        spark.stop()
+      }
+    }
+)
+
+object ChangesetORCUpdaterUtils {
+  implicit val readInstant: Argument[Instant] = new Argument[Instant] {
+    override def read(string: String): ValidatedNel[String, Instant] = {
+      try { Validated.valid(Instant.parse(string)) }
+      catch { case e: Exception => Validated.invalidNel(s"Invalid time: $string (${ e.getMessage })") }
+    }
+
+    override def defaultMetavar: String = "time"
+  }
+
+  private val formatter = DateTimeFormat.forPattern("y-M-d H:m:s.SSSSSSSSS Z")
+
+  private implicit val dateTimeDecoder: Decoder[DateTime] =
+    Decoder.instance(a => a.as[String].map(DateTime.parse(_, formatter)))
+
+  case class Sequence(last_run: DateTime, sequence: Long)
+
+  def getCurrentSequence(baseURI: URI): Sequence = {
```

Comment:
I like this signature more than VP's current `def getCurrentSequence(baseURI: URI): Option[Int]`. Returning `Option[Sequence]` instead of throwing if it can't be determined would be nice though.

Reply:
Note https://github.com/geotrellis/vectorpipe/blob/d4c5d36430be79cb438dcc38954bb28adc9a7901/src/main/scala/vectorpipe/sources/ChangesetSource.scala lines 73 and 95